src/auth/controller/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  Post,
+  Request,
+  UnauthorizedException,
+  UseGuards,
+} from '@nestjs/common';
+import { AuthService } from '../service/auth.service';
+import { RefreshJwtGuard } from '../guard/refresh-jwt-auth.guard';
+import { ApiBody } from '@nestjs/swagger';
+import { User } from '../domain/user.domain';
+import { RefreshToken } from '../domain/refresh-token.domain';
+import { LocalAuthGuard } from '../guard/local-auth.guard';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private authService: AuthService) {}
+
+  @UseGuards(LocalAuthGuard)
+  @Post('login')
+  @ApiBody({ type: User })
+  async login(@Request() req: { body: User }) {
+    if (!req.body.username || !req.body.password) {
+      throw new UnauthorizedException(
+        'You must provide a username and a password!',
+      );
```

Comment:
You could add this check inside the service method where you already have another checks. Same for other methods.

No need to change this now, it's fine.

Reply:
done